core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -534,6 +534,19 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
   const Policy m_rp;
 
  public:
+  template <typename Policy, typename Functor>
+  inline static int max_tile_size_product(const Policy& pol, const Functor&) {
```

Comment:
You don't need inline here, because it is an in-class definition (and this isn't consistent throughout the file).

Reply:
I'll do this for the other backends, too